src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -314,6 +291,21 @@ private static SslClientAuthenticationOptions ConstructSslOptions(HttpConnection
             }
         }
 
+        private async ValueTask<(HttpConnectionBase connection, bool isNewConnection, HttpResponseMessage failureResponse)>
+            WaitForAvailableHttp11Connection(TaskCompletionSource<HttpConnection> waiterTaskSource, HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            using (cancellationToken.Register(s => ((TaskCompletionSource<HttpConnection>)s).TrySetCanceled(), waiterTaskSource))
```

Comment:
We could decide this doesn't matter, but you'll notice the previous TrySetCanceled call was passing in the CancellationToken as an argument.  That's so that the OperationCanceledException that propagated out would contain that CancellationToken, so someone could potentially compare it against the one they provide.  To fix that here we'd either need to allocate or to again derive from the TCS in order to store the CT onto it (as we had before).

Reply:
Do you think it's important? It seems straightforward to derive from TCS again if we need to.

Reply:
It's very corner case. But I think I'll sleep better if we just do it.

Reply:
Ok, I'll make it work.

Side note: Might be nice if CancellationToken.Register passed the CancellationToken to the register callback...